azurerm/internal/services/maintenance/maintenance_configuration_resource.go

```diff
@@ -49,9 +49,7 @@ func resourceArmMaintenanceConfiguration() *schema.Resource {
 
 			"location": azure.SchemaLocation(),
 
-			// There's a bug in the Azure API where this is returned in lower-case
-			// BUG: https://github.com/Azure/azure-rest-api-specs/issues/8653
-			"resource_group_name": azure.SchemaResourceGroupNameDiffSuppress(),
+			"resource_group_name": azure.SchemaResourceGroupName(),
```

Comment:
This'll only work for new resources and not existing ones - which'll need a state migration to correct the casing of the resource group name (by queuing that API if that's the case?

Reply:
I thought there is no need to change the `resource_group_name`, so I just revert it

Reply:
should we mark this as a todo for 3.0 and make that breaking change then?